indy_common/auth.py

```diff
@@ -67,7 +67,8 @@ def generate_auth_map(valid_roles, need_permission=None):
         '{}_<any>_<any>_<any>'.format(VALIDATOR_INFO):
             {TRUSTEE: [], STEWARD: []},
     }
-    if need_permission is False or (need_permission is None and not getConfig().WRITES_REQUIRE_TRUST_ANCHOR):
+    if anyone_can_write is True or (anyone_can_write is None and getConfig().ANYONE_CAN_WRITE):
+        auth_map['{}_role__'.format(NYM)][None] = []
```

Comment:
This rule is only about creation of new NYMs with empty role (common users). What about other roles? We need to clarify the requirement.

Reply:
If any NYMs, SCHEMAs, CLAIM_DEFs, and revocation txns can be created with ANYONE_CAN_WRITE=True, then maybe we should not modify this map, but rather add a check to `authorised`, so that if `typ` is one from the list above, then return True immediately.

Reply:
asked